src/_pytest/nodes.py

```diff
@@ -594,7 +598,7 @@ def __init__(
             try:
                 nodeid = str(self.path.relative_to(session.config.rootpath))
             except ValueError:
-                nodeid = _check_initialpaths_for_relpath(session, path)
+                nodeid = _check_initialpaths_for_relpath(session._initialpaths, path)
```

Comment:
@bluetech @nicoddemus seeing this right in place i'm thinking - what if they cooperated with the parent

with the introduction of directory nodes i believe the case is a lot more clear cut and possibly could avoid the common path calls all-together

also it seems like going for directory names more times could potentially safe a lot of time as commonly there are many files in the same directory

Reply:
> what if they cooperated with the parent with the introduction of directory nodes i believe the case is a lot more clear cut and possibly could avoid the common path calls all-together

Not sure I follow, we already have `Directory` nodes:

https://github.com/pytest-dev/pytest/blob/cfbe319d0c98302eae507622da8908be351fed2e/src/_pytest/nodes.py#L635